src/main/java/org/mule/extension/email/internal/value/MailboxFolderValueProvider.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extension.email.internal.value;
+
+import static org.mule.runtime.extension.api.values.ValueBuilder.getValuesFor;
+
+import org.mule.extension.email.internal.mailbox.MailboxConnection;
+import org.mule.runtime.api.value.Value;
+import org.mule.runtime.extension.api.annotation.param.Connection;
+import org.mule.runtime.extension.api.values.ValueProvider;
+import org.mule.runtime.extension.api.values.ValueResolvingException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+import javax.mail.Folder;
+
+/**
+ * {@link ValueProvider} implementation which provides the list of {@link Folder} on the user namespace.
+ * 
+ * @since 1.1.0
+ */
+public class MailboxFolderValueProvider implements ValueProvider {
+
+  @Connection
+  private MailboxConnection mailboxConnection;
+
+  @Override
+  public Set<Value> resolve() throws ValueResolvingException {
+    Folder[] folders = mailboxConnection.listFolders();
+
+    List<String> foldersNames = new ArrayList<>();
```

Comment:
You could simplify your code with something more functional: `return getValuesFor(stream(folders).map(Folder::getFullName));`